fastify.js

```diff
@@ -387,6 +391,13 @@ function build (options) {
     })
   }
 
+  function onRouteIterator (fn, arg, done) {
+    const result = fn(arg, done)
+    if (result && typeof result.then === 'function') {
+      result.then(result => done(null)).catch(done)
+    }
+  }
```

Comment:
I think we should have a similar function to this one, don't we?

Reply:
No, but I could create a helper method.